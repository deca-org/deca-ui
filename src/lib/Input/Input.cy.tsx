import { Test } from '../Utils';
import _cyp from '../../../cypress';
import React from 'react';
import Input from './Input';
import { standardColors } from '../Theme';

describe('components/Input', () => {
  it('input control', () => {
    cy.mount(
      <Input
        variant="solid"
        placeholder="Example"
        label="Label"
        helperText="Helper text"
      />
    );
    cy.get('input').type('Hello, World');
    cy.get('input').should('have.value', 'Hello, World');
  });
  describe('solid variant', () => {
    it('background-color', () => {
      cy.mount(
        <Input
          variant="solid"
          placeholder="Example"
          label="Label"
          helperText="Helper text"
        />
      );
      cy.get('.decaInput-wrap').should(
        'have.css',
        'background-color',
        Test.color('gray200')
      );
      cy.get('input').click();
      cy.get('.decaInput-wrap').should(
        'have.css',
        'background-color',
        Test.color('gray300')
      );
    });
    it('disabled state', () => {
      cy.mount(
        <Input
          disabled
          variant="solid"
          placeholder="Example"
          label="Label"
          helperText="Helper text"
        />
      );
      cy.get('.decaInput-label').should(
        'have.css',
        'color',
        Test.color('gray500')
      );

      cy.get('.decaInput-wrap').should(
        'have.css',
        'background-color',
        Test.color('gray200')
      );

      cy.get('.decaInput-wrap').should('have.css', 'cursor', 'not-allowed');

      cy.get('.decaInput-helperText').should(
        'have.css',
        'color',
        Test.color('gray500')
      );
    });
  });
  describe('outlined variant', () => {
    it('transparent background-color', () => {
      cy.mount(
        <Input
          variant="outlined"
          placeholder="Example"
          label="Label"
          helperText="Helper text"
        />
      );
      cy.get('input').click();
      cy.get('.decaInput-wrap').should(
        'have.css',
        'background-color',
        'rgba(0, 0, 0, 0)'
      );
    });
    describe('focus color', () => {
      standardColors.map((color) => {
        it(color, () => {
          cy.mount(
            <Input
              variant="outlined"
              focusColor={color}
              placeholder="Example"
              label="Label"
              helperText="Helper text"
            />
          );
          cy.get('.decaInput-wrap').should(
            'have.css',
            'color',
            Test.color('black')
          );
          cy.get('input').click();
          cy.get('.decaInput-wrap').should(
            'have.css',
            'border-color',
            Test.color(color)
          );
          cy.get('.decaInput-label').should(
            'have.css',
            'color',
            Test.color(color)
          );
        });
      });
    });
    it('disabled state', () => {
      cy.mount(
        <Input
          disabled
          variant="outlined"
          placeholder="Example"
          label="Label"
          helperText="Helper text"
        />
      );
      cy.get('.decaInput-label').should(
        'have.css',
        'color',
        Test.color('gray500')
      );

      cy.get('.decaInput-wrap').should(
        'have.css',
        'border-color',
        Test.color('gray400')
      );

      cy.get('.decaInput-wrap').should('have.css', 'cursor', 'not-allowed');

      cy.get('.decaInput-helperText').should(
        'have.css',
        'color',
        Test.color('gray500')
      );
    });
  });
  describe('dark mode', () => {
    it('label & helperText color', () => {
      cy.darkMount(
        <Input placeholder="Example" label="Label" helperText="Helper text" />
      );
      cy.get('.decaInput-label').should(
        'have.css',
        'color',
        Test.color('white')
      );

      cy.get('.decaInput-helperText').should(
        'have.css',
        'color',
        Test.color('white')
      );
    });
    describe('solid variant', () => {
      it('background-color', () => {
        cy.darkMount(
          <Input
            variant="solid"
            placeholder="Example"
            label="Label"
            helperText="Helper text"
          />
        );
        cy.get('.decaInput-wrap').should(
          'have.css',
          'background-color',
          Test.color('gray900')
        );
        cy.get('input').click();
        cy.get('.decaInput-wrap').should(
          'have.css',
          'background-color',
          Test.color('gray800')
        );
      });
      it('disabled state', () => {
        cy.darkMount(
          <Input
            disabled
            variant="solid"
            placeholder="Example"
            label="Label"
            helperText="Helper text"
          />
        );

        cy.get('.decaInput-label').should(
          'have.css',
          'color',
          Test.color('gray700')
        );

        cy.get('.decaInput-wrap').should(
          'have.css',
          'background-color',
          Test.color('gray900')
        );

        cy.get('.decaInput-wrap').should('have.css', 'cursor', 'not-allowed');

        cy.get('.decaInput-helperText').should(
          'have.css',
          'color',
          Test.color('gray700')
        );
      });
    });
    describe('outlined variant', () => {
      it('disabled state', () => {
        cy.darkMount(
          <Input
            disabled
            variant="outlined"
            placeholder="Example"
            label="Label"
            helperText="Helper text"
          />
        );
        cy.get('.decaInput-label').should(
          'have.css',
          'color',
          Test.color('gray700')
        );

        cy.get('.decaInput-wrap').should(
          'have.css',
          'border-color',
          Test.color('gray800')
        );

        cy.get('.decaInput-wrap').should('have.css', 'cursor', 'not-allowed');

        cy.get('.decaInput-helperText').should(
          'have.css',
          'color',
          Test.color('gray700')
        );
      });
    });
  });
});
