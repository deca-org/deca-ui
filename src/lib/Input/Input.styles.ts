import { styled } from '@lib/Theme/stitches.config';

export const StyledInputMainContainer = styled('div', {
  width: '100%',
  fontFamily: '$normal',
  compoundVariants: [
    {
      size: 'sm',
      maxWidth: false,
      css: {
        width: '$32',
      },
    },
    {
      size: 'md',
      maxWidth: false,
      css: {
        width: '$35',
      },
    },
    {
      size: 'lg',
      maxWidth: false,
      css: {
        width: '$35',
      },
    },
  ],
  variants: {
    maxWidth: {
      true: {
        width: '100%',
      },
      false: {},
    },
    isDisabled: {
      true: {
        cursor: 'not-allowed',
      },
      false: {},
    },
    size: {
      sm: {},
      md: {},
      lg: {},
    },
  },
});

export const StyledInputLabel = styled('label', {
  display: 'inline-block',
  fontWeight: '$semibold',
  '-webkit-font-smoothing': 'antialiased',
  transition: '$default',
  ml: '$0',
  mb: '$2',
  mt: '0px',
  color: '$text',
  compoundVariants: [
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'primary',
      css: {
        color: '$primary',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'secondary',
      css: {
        color: '$secondary',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'success',
      css: {
        color: '$success',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'warning',
      css: {
        color: '$warning',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'error',
      css: {
        color: '$error',
      },
    },
    {
      variant: 'outlined',
      state: 'default',
      css: {
        color: '$text',
      },
    },
    {
      variant: 'outlined',
      state: 'value',
      css: {
        color: '$text',
      },
    },
    {
      state: 'isDisabled',
      isDark: 'true',
      css: {
        color: '$gray700',
      },
    },
  ],
  variants: {
    size: {
      sm: {
        mb: '$1',
        fontSize: '$footnote',
      },
      md: {
        fontSize: '$caption',
      },
      lg: {
        fontSize: '$caption',
      },
    },
    state: {
      focused: {},
      isDisabled: {
        color: '$gray500',
      },
      value: {},
      default: {},
    },
    variant: {
      solid: {},
      outlined: {},
    },
    focusColor: {
      primary: {},
      secondary: {},
      success: {},
      warning: {},
      error: {},
    },
    pill: {
      true: {
        ml: '$2',
      },
      false: {},
    },
    isDark: {
      true: {},
      false: {},
    },
  },
});

export const StyledInputHelperText = styled('p', {
  fontSize: '$footnote',
  color: '$gray600',
  lineHeight: '$0',
  transition: '$default',
  m: '$n',
  ml: '$0',
  mt: '$2',
  compoundVariants: [
    {
      state: 'isDisabled',
      isDark: true,
      css: {
        color: '$gray700',
      },
    },
    {
      state: 'value',
      isDark: true,
      css: {
        color: '$text',
      },
    },
    {
      state: 'default',
      isDark: true,
      css: {
        color: '$text',
      },
    },
  ],
  variants: {
    size: {
      sm: {
        mt: '$1',
      },
      md: {},
      lg: {},
    },
    state: {
      focused: {},
      isDisabled: {
        color: '$gray500',
      },
      value: {
        color: '$gray600',
      },
      default: {
        color: '$gray600',
      },
    },
    pill: {
      true: {
        ml: '$2',
      },
      false: {},
    },
    isDark: {
      true: {
        color: '$text',
      },
      false: {},
    },
  },
});

export const StyledInputContainer = styled('div', {
  fontWeight: '$medium',
  boxSizing: 'border-box',
  borderWidth: '$normal',
  borderStyle: 'solid',
  transition: '$default',
  compoundVariants: [
    {
      variant: 'solid',
      isDark: false,
      css: {
        backgroundColor: '$gray200',
        borderColor: '$gray200',
      },
    },
    {
      variant: 'solid',
      state: 'focused',
      isDark: false,
      css: {
        backgroundColor: '$gray300',
        borderColor: '$gray300',
      },
    },
    {
      variant: 'solid',
      isDark: true,
      css: {
        backgroundColor: '$gray900',
        borderColor: '$gray900',
      },
    },
    {
      variant: 'solid',
      state: 'focused',
      isDark: true,
      css: {
        backgroundColor: '$gray800',
        borderColor: '$gray800',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'primary',
      css: {
        borderColor: '$primary',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'secondary',
      css: {
        borderColor: '$secondary',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'success',
      css: {
        borderColor: '$success',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'warning',
      css: {
        borderColor: '$warning',
      },
    },
    {
      variant: 'outlined',
      state: 'focused',
      focusColor: 'error',
      css: {
        borderColor: '$error',
      },
    },
    {
      variant: 'outlined',
      state: 'isDisabled',
      css: {
        borderColor: '$gray400',
      },
    },
    {
      variant: 'outlined',
      state: 'value',
      css: {
        borderColor: '$gray400',
      },
    },
    {
      variant: 'outlined',
      state: 'default',
      css: {
        borderColor: '$gray400',
      },
    },
    {
      variant: 'outlined',
      state: 'value',
      isDark: true,
      css: {
        borderColor: '$gray700',
      },
    },
    {
      variant: 'outlined',
      state: 'default',
      isDark: true,
      css: {
        borderColor: '$gray700',
      },
    },
    {
      variant: 'outlined',
      state: 'isDisabled',
      isDark: true,
      css: {
        borderColor: '$gray800',
      },
    },
    {
      variant: 'solid',
      state: 'isDisabled',
      isDark: true,
      css: {
        borderColor: '$gray900',
      },
    },
    {
      size: 'sm',
      maxWidth: false,
      css: {
        width: '$32',
      },
    },
    {
      size: 'md',
      maxWidth: false,
      css: {
        width: '$35',
      },
    },
    {
      size: 'lg',
      maxWidth: false,
      css: {
        width: '$35',
      },
    },
  ],
  variants: {
    maxWidth: {
      true: {
        width: '100%',
      },
      false: {},
    },
    size: {
      sm: {
        height: '$6',
      },
      md: {
        height: '$9',
      },
      lg: {
        height: '$10',
      },
    },
    state: {
      focused: {},
      isDisabled: {
        cursor: 'not-allowed',
      },
      value: {},
      default: {},
    },
    variant: {
      solid: {},
      outlined: {},
    },
    focusColor: {
      primary: {},
      secondary: {},
      success: {},
      warning: {},
      error: {},
    },
    pill: {
      true: {
        borderRadius: '$pill',
      },
      false: {
        borderRadius: '$sm',
      },
    },
    isDark: {
      true: {},
      false: {},
    },
  },
});

export const StyledInput = styled('input', {
  color: '$text',
  bg: '$transparent',
  border: 'none',
  '&:focus': {
    outline: 0,
  },
  boxSizing: 'border-box',
  size: '100%',
  paddingRight: 'calc($2 * 1.5)',
  paddingLeft: 'calc($2 * 1.5)',
  transition: '$default',
  '&::placeholder': {
    color: '$gray700',
    opacity: 0.75,
  },
  compoundVariants: [
    {
      variant: 'solid',
      isDark: false,
      css: {
        backgroundColor: '$gray200',
      },
    },
    {
      variant: 'solid',
      state: 'focused',
      isDark: false,
      css: {
        backgroundColor: '$gray300',
        borderColor: '$gray300',
      },
    },
    {
      variant: 'solid',
      isDark: true,
      css: {
        backgroundColor: '$gray900',
      },
    },
    {
      variant: 'solid',
      state: 'focused',
      isDark: true,
      css: {
        backgroundColor: '$gray800',
        borderColor: '$gray800',
      },
    },
    {
      variant: 'outlined',
      state: 'isDisabled',
      isDark: true,
      css: {
        '&::placeholder': {
          color: '$gray600',
        },
        cursor: 'not-allowed',
      },
    },
    {
      variant: 'solid',
      state: 'isDisabled',
      isDark: true,
      css: {
        '&::placeholder': {
          color: '$gray700',
        },
        cursor: 'not-allowed',
      },
    },
  ],
  variants: {
    variant: {
      solid: {},
      outlined: {},
    },
    size: {
      sm: {
        fontSize: '$footnote',
      },
      md: {
        fontSize: '$bodySm',
      },
      lg: {
        fontSize: '$bodySm',
      },
    },
    state: {
      focused: {},
      isDisabled: {
        '&::placeholder': {
          color: '$gray500',
        },
        cursor: 'not-allowed',
      },
      value: {},
      default: {},
    },
    pill: {
      true: {
        px: '$3',
        borderRadius: '$pill',
      },
      false: {
        borderRadius: '$sm',
      },
    },
    isDark: {
      true: {
        '&::placeholder': {
          color: '$gray100',
        },
      },
      false: {},
    },
  },
});
