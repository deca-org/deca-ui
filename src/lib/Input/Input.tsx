import { ThemeContext } from '@lib/Theme';
import { CSS, StandardColors } from '@lib/Theme/stitches.config';
import {
  Modify,
  uuid,
  __DEV__,
  PolymorphicRef,
  PolymorphicComponentPropWithRef,
} from '@lib/Utils';
import clsx from 'clsx';
import React, { useMemo, useState } from 'react';

import {
  StyledInputMainContainer,
  StyledInputLabel,
  StyledInputHelperText,
  StyledInputContainer,
  StyledInput,
} from './Input.styles';

/**
 * Type for Input and Text Area elements
 */
export type FormElement = HTMLInputElement | HTMLTextAreaElement;

/**
 * Inputs allow users to enter text into a UI. They typically appear in forms and dialogs.
 */
interface Props
  extends Modify<
    React.ComponentPropsWithRef<'input'>,
    {
      /**
       * Size of the component.
       * @default md
       */
      size?: 'sm' | 'md' | 'lg';
    }
  > {
  /**
   * Size of the component.
   * @default md
   */
  size?: 'sm' | 'md' | 'lg';
  /**
   * Text label for the component.
   */
  label?: string;
  /**
   * ClassName applied to the component.
   * @default ''
   */
  className?: string;
  /**
   * Disabled state applied to the component.
   * @default false
   */
  disabled?: boolean;
  /**
   * The variant to use.
   * @default solid
   */
  variant?: 'solid' | 'outlined';
  /**
   * Placeholder text for component.
   */
  placeholder?: string;
  /**
   * Override default CSS style
   */
  css?: CSS;
  /**
   * Color of text and border when input is focused.
   * @default primary
   */
  focusColor?: StandardColors;
  /**
   * Set input width to 100%.
   * @default false
   */
  maxWidth?: boolean;
  /**
   * Required input prop.
   * @default false
   */
  required?: boolean;
  /**
   * Add helper text to input.
   * @default ''
   */
  helperText?: string;
  /**
   * The current value of the input.
   */
  value?: string;
  /**
   * The initial value of the input.
   */
  initialValue?: string;
  /**
   * Function that executes when input is changed.
   */
  onChange?(e: React.ChangeEvent<FormElement>): void;
  /**
   * Function that executes when input is focused.
   */
  onFocus?(e: React.FocusEvent<FormElement>): void;
  /**
   * Function that executes when input is just out of focus.
   */
  onBlur?(e: React.FocusEvent<FormElement>): void;
  /**
   * Have input be pill shaped
   */
  pill?: boolean;
}

export type InputProps<T extends React.ElementType> =
  PolymorphicComponentPropWithRef<T, Props>;

export type InputComponent = (<C extends React.ElementType = 'input'>(
  props: InputProps<C>
) => React.ReactElement | null) & { displayName?: string };

const Input: InputComponent = React.forwardRef(
  <T extends React.ElementType = 'input'>(
    {
      label,
      className = '',
      disabled = false,
      size = 'md',
      variant = 'solid',
      as,
      css,
      focusColor = 'primary',
      maxWidth = false,
      placeholder,
      required = false,
      helperText = '',
      value,
      initialValue = '',
      onChange,
      onFocus,
      onBlur,
      pill = false,
      ...props
    }: InputProps<T>,
    ref?: PolymorphicRef<T>
  ) => {
    const [selfValue, setSelfValue] = useState<string>(initialValue);
    const [focused, setFocused] = useState<boolean>(false);

    const isControlledComponent = useMemo(() => value !== undefined, [value]);

    const changeHandler = (e: React.ChangeEvent<FormElement>) => {
      if (disabled) return;
      if (!isControlledComponent) {
        setSelfValue(e.target.value);
      }
      onChange && onChange(e);
    };

    const focusHandler = (e: React.FocusEvent<FormElement>) => {
      if (disabled) return;
      setFocused(true);
      onFocus && onFocus(e);
    };

    const blurHandler = (e: React.FocusEvent<FormElement>) => {
      if (disabled) return;
      setFocused(false);
      onBlur && onBlur(e);
    };

    const getState = useMemo(() => {
      return focused
        ? 'focused'
        : disabled
        ? 'isDisabled'
        : selfValue || value
        ? 'value'
        : 'default';
    }, [focused, disabled, selfValue, value]);

    const inputId = uuid('input');

    const preClass = 'decaInput';

    const { dark } = React.useContext(ThemeContext);

    return (
      <StyledInputMainContainer
        className={clsx(className, `${preClass}-root`)}
        css={css}
        size={size}
        maxWidth={maxWidth}
      >
        {label && (
          <StyledInputLabel
            size={size}
            state={getState}
            className={`${preClass}-label`}
            variant={variant}
            focusColor={focusColor}
            htmlFor={inputId}
            pill={pill}
            isDark={dark}
          >
            {label}
          </StyledInputLabel>
        )}
        <StyledInputContainer
          size={size}
          state={getState}
          className={`${preClass}-wrap`}
          variant={variant}
          focusColor={focusColor}
          maxWidth={maxWidth}
          pill={pill}
          isDark={dark}
        >
          <StyledInput
            onFocus={focusHandler}
            onBlur={blurHandler}
            onChange={changeHandler}
            className={`${preClass}-input`}
            ref={ref}
            variant={variant}
            state={getState}
            as={as}
            size={size}
            value={isControlledComponent ? value : selfValue}
            required={required}
            disabled={disabled}
            placeholder={placeholder}
            id={inputId}
            pill={pill}
            isDark={dark}
            {...props}
          ></StyledInput>
        </StyledInputContainer>
        {helperText && (
          <StyledInputHelperText
            size={size}
            state={getState}
            className={`${preClass}-helperText`}
            pill={pill}
            isDark={dark}
          >
            {helperText}
          </StyledInputHelperText>
        )}
      </StyledInputMainContainer>
    );
  }
);

if (__DEV__) {
  Input.displayName = 'DecaUI.Input';
}

export default Input;
