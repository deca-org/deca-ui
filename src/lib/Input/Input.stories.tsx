import { DecaUIProvider } from '@lib/Theme';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import Input from './Input';

export default {
  title: 'Input',
  component: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const Solid = Template.bind({});

Solid.args = {
  label: 'Email Address',
  size: 'lg',
  helperText: 'Please submit query',
  variant: 'solid',
  placeholder: 'e.g. johndoe@gmail.com',
  focusColor: 'primary',
  required: true,
  disabled: false,
  as: 'input',
  maxWidth: false,
  initialValue: '',
  className: '',
  pill: false,
};

export const Outlined = Template.bind({});

Outlined.args = {
  label: 'Email Address',
  size: 'lg',
  helperText: 'Please submit query',
  variant: 'outlined',
  focusColor: 'primary',
  placeholder: 'e.g. johndoe@gmail.com',
  required: true,
  disabled: false,
  as: 'input',
  maxWidth: false,
  initialValue: '',
  className: '',
  pill: false,
};

export const WithTheme = Template.bind({});

WithTheme.args = { ...Outlined.args };
WithTheme.decorators = [
  (Story) => (
    <DecaUIProvider
      theme={{
        colors: {
          primary: '$green600',
        },
        radii: {
          sm: '15px',
        },
      }}
    >
      <Story />
    </DecaUIProvider>
  ),
];

export const DarkMode = Template.bind({});

DarkMode.args = { ...Outlined.args };
DarkMode.parameters = { backgrounds: { default: 'dark' } };

DarkMode.decorators = [
  (Story) => (
    <DecaUIProvider mode="dark">
      <Story />
    </DecaUIProvider>
  ),
];
