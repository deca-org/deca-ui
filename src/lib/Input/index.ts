import Input from './Input';
export * from './Input';
export { Input };
export default Input;
