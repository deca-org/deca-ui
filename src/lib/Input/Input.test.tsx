import { render, fireEvent, act } from '@testing-library/react';
import React from 'react';

import Input from './Input';

describe('components/Input', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Input
        helperText="Helper Text"
        label="Label Text"
        placeholder="Placeholder Text"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('onFocus event fires', () => {
    const mockFn = jest.fn();
    const utils = render(<Input label="Label Text" onFocus={mockFn} />);
    const input = utils.getByLabelText('Label Text');
    act(() => {
      input.focus();
    });
    expect(mockFn.mock.calls.length).toBe(1);
  });

  it('renders all colors', () => {
    const { asFragment } = render(
      <>
        <Input focusColor="primary" />
        <Input focusColor="secondary" />
        <Input focusColor="success" />
        <Input focusColor="warning" />
        <Input focusColor="error" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders all sizes', () => {
    const { asFragment } = render(
      <>
        <Input size="sm" />
        <Input size="md" />
        <Input size="lg" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders all variants', () => {
    const { asFragment } = render(
      <>
        <Input variant="solid" />
        <Input variant="outlined" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should ignore events when disabled', () => {
    const mockFn = jest.fn();
    const utils = render(<Input disabled label="Label Text" />);
    const input = utils.getByLabelText('Label Text');
    fireEvent.change(input, { target: { value: 'new-value' } });
    expect(mockFn.mock.calls.length).toBe(0);
  });
});
