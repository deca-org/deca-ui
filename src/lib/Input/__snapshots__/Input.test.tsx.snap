// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <label
      class="c-fdMnTA c-fdMnTA-cNhXCf-size-md decaInput-label"
      for="input-1"
    >
      Label Text
    </label>
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        placeholder="Placeholder Text"
        value=""
      />
    </div>
    <p
      class="c-bdULyd c-bdULyd-RRNal-state-default decaInput-helperText"
    >
      Helper Text
    </p>
  </div>
</DocumentFragment>
`;

exports[`components/Input renders all colors 1`] = `
<DocumentFragment>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Input renders all sizes 1`] = `
<DocumentFragment>
  <div
    class="c-dxhRnB c-dxhRnB-llZTNE-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-jigqbG-size-sm c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-llZTNE-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-kGUnSj-size-sm c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-dBFJVN-size-lg c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-lg c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Input renders all variants 1`] = `
<DocumentFragment>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-htQczz-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false c-dlVhqe-fOmxiH-cv decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
  <div
    class="c-dxhRnB c-dxhRnB-dbfLtX-cv decaInput-root"
  >
    <div
      class="c-hQCSiP c-hQCSiP-hlHOLx-size-md c-hQCSiP-cycqzJ-pill-false c-hQCSiP-dBEtCv-cv c-hQCSiP-dbfLtX-cv decaInput-wrap"
    >
      <input
        class="c-dlVhqe c-dlVhqe-bxpJOv-size-md c-dlVhqe-cycqzJ-pill-false decaInput-input"
        id="input-1"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
